include/h2/gpu/rocm/memory_utils.hpp

```diff
@@ -13,24 +13,22 @@
  *  Thin wrappers around hipMem{cpy,set} functions. These are here so
  *  they can be inlined if possible.
  */
+#if H2_HAS_ROCM
+#include "hydrogen/PoolAllocator.hpp"
+#endif
```

Comment:
```suggestion
#include <hydrogen/PoolAllocator.hpp>
```
This file shouldn't be included unless `H2_HAS_ROCM`.